core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -51,8 +54,39 @@ private[spark] class TaskDescription(
     val index: Int,    // Index within this task's TaskSet
     val addedFiles: Map[String, Long],
     val addedJars: Map[String, Long],
-    val properties: Properties,
-    val serializedTask: ByteBuffer) {
+    val properties: Properties) extends Logging {
+
+  def this(
+      taskId: Long,
+      attemptNumber: Int,
+      executorId: String,
+      name: String,
+      index: Int, // Index within this task's TaskSet
+      addedFiles: Map[String, Long],
+      addedJars: Map[String, Long],
+      properties: Properties,
+      task: Task[_]) {
+      this(taskId, attemptNumber, executorId, name, index,
+        addedFiles, addedJars, properties)
+      task_ = task
```

Comment:
I think TaskDescription has two constructors that are more readable, and we can add a comment about the constructor with the `task: Task [_]` parameter is only be called in driver.